src/Microsoft.ML/MemoryCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML
+{
+    public class MemoryCollection
```

Comment:
>MemoryCollection [](start = 17, length = 16)

Is this a static class? Why does it exist at all?  Seems like a very strange name for a class

Reply:
It's a helper class, see comment from Tom starting from "Just an observation".

---
In reply to: [187222326](https://github.com/dotnet/machinelearning/pull/106#discussion_r187222326) [](ancestors = 187222326)

Reply:
Can't you have that method on the MemoryCollectionLoader?

---
In reply to: [187222607](https://github.com/dotnet/machinelearning/pull/106#discussion_r187222607) [](ancestors = 187222607,187222326)

Reply:
MemoryCollectionLoader is a generic class. Whole point to not type class twice. MemoryCollectionLoader<T>.Create(List<T>data) and MemoryCollection.Create(List<T>data).

---
In reply to: [187223070](https://github.com/dotnet/machinelearning/pull/106#discussion_r187223070) [](ancestors = 187223070,187222607,187222326)

Reply:
Why do we need the create method anyway? What's wrong with a good old constructor?

---
In reply to: [187223445](https://github.com/dotnet/machinelearning/pull/106#discussion_r187223445) [](ancestors = 187223445,187223070,187222607,187222326)

Reply:
For problem with "good old constructor," see earlier comment from the aforementioned Tom on "Just an observation." The more we can rely on the compiler to do the work for the user, the better, I'd say. Especially since this is merely a wrapper *anyway*, I don't see the point of making the user's job any harder than it has to be.

---
In reply to: [187226013](https://github.com/dotnet/machinelearning/pull/106#discussion_r187226013) [](ancestors = 187226013,187223445,187223070,187222607,187222326)

Reply:
If we must, then lest's just create a non-generic with the same name as the same name as the generic class, make it static and have those create methods.  No need to pollute namespace with confusing names.

---
In reply to: [187230636](https://github.com/dotnet/machinelearning/pull/106#discussion_r187230636) [](ancestors = 187230636,187226013,187223445,187223070,187222607,187222326)

Reply:
Honestly I'd rather just not expose the implementations at all, have the static create methods return an object of the appropriate interface.

---
In reply to: [187240158](https://github.com/dotnet/machinelearning/pull/106#discussion_r187240158) [](ancestors = 187240158,187230636,187226013,187223445,187223070,187222607,187222326)

Reply:
>>I'd rather just not expose the implementations at all, have the static create methods return an object of the appropriate interface
Done.

---
In reply to: [187352180](https://github.com/dotnet/machinelearning/pull/106#discussion_r187352180) [](ancestors = 187352180,187240158,187230636,187226013,187223445,187223070,187222607,187222326)